apps/backend/app/services/s3_service.py

```
@@ -120,15 +120,36 @@ def download_file_from_s3(s3_url: str) -> str:
+            logger.warning("S3Service initialized in mock mode (test credentials or missing credentials)")
+            # Create a dummy client that won't be used
+            self.s3_client = None
+        else:
+            try:
+                self.s3_client = boto3.client(
+                    "s3",
+                    aws_access_key_id=aws_access_key,
+                    aws_secret_access_key=aws_secret_key,
+                    region_name=os.getenv("AWS_REGION", "us-east-1"),
+                )
+            except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)